# Problema 1

Considere la sucesión $a_{n} = 3n^{2} + 7$. Por medio de un código en *Python* despliega en pantalla:

1. Los primeros $5$ términos.
2. Los $5$ términos que vienen inmediatamente después del decimoquinto término.

# Problema 2

Considere la sucesión $g_{n} = 5n^{3}$. Implemente un código en *Python* que muestre en pantalla:

1. Los primeros $4$ términos de la sucesión.
2. Los $4$ términos que vienen inmediatamente después del octavo término.
3. Determinar si el término $40.000$ pertenece a la sucesión. Si es así, indicar en qué posición se encuentra.

# Problema 3

La sucesión de Fibonacci es una serie de números en la que cada número, desde el tercero en adelante, es la suma de los dos anteriores. Comienza con el $0$ y $1$, tal como se observa a continuación:

$$0, 1, 1, 2, 3, 5, 8, 13, ...$$

Formalmente, la sucesión se define de la siguiente manera:

\begin{split}
F(0) &= 0\\
F(1) &= 1\\
F(n) &= F(n - 1) + F(n - 2), \ \text{ para $n \geq 2$}
\end{split}

Esta sucesión se utiliza en una variedad de contextos en informática y algoritmos debido a sus propiedades matemáticas y patrones.

1. Implementa un código en *Python* que muestre en pantalla los primeros $20$ términos de la sucesión.
2. Calcula, utilizando un ciclo *for*, la suma de los primeros $30$ términos de la sucesión de Fibonacci.

# Problema 4

Implementa un código que permita guardar en una lista la sucesión de los primeros $n$ números pares, con $n \geq 1$. Luego, determina:

1. ¿Cuál es el valor del décimo término?
2. Calcula, utilizando un ciclo *for*, la suma de los primeros $100$ números pares.
3. ¿En qué posición se encuentra el número $58$ dentro de la sucesión?

# Problema 5

Un nuevo software tiene $500$ usuarios y se espera que el número de usuarios crezca un $15\%$ respecto al mes anterior. Considere que $b_{n}$ es el número de usuarios transcurridos $n$ meses.

1. Calcula cuántos usuarios tendrá el software al cabo de tres meses. ¿Y al cabo de seis meses?
2. Mediante un código en *Pyhton* muestra la cantidad de usuarios que tendrá el software cada mes durante el primer año.
2. ¿Cuántos usuarios tendrá en total el software luego de un año? Utilice ciclos *for* para calcular.

# Problema 6

Un estudiante de ingeniería informática planea ahorrar para un proyecto personal. Para esto, el primer mes deposita en una cuenta $\$12.000$, el segundo mes deposita $\$14.000$ y así sucesivamente, de modo que cada mes aumenta el depósito en $\$2.000$. Considere que $a_{i}$ corresponde al depósito del estudiante en el mes $i$.

1. Implementa un código que permita mostrar en pantalla lo que depositó mensualmente durante el primer año.
2. ¿Cuánto depositó el estudiante en febrero del segundo año?
3. ¿Cuál será el total ahorrado al finalizar dos años?

#Creación de sucesiones con listas
sucesion = []
for i in range(cantidad de términos a mostrar):
    n = i + 1
    sucesion.append(Término general)
    print(f'El término {n} es {sucesion[i]}')

In [2]:
# Creación de sucesiones con listas

sucesion = []

for i in range(12):  # cantidad de términos a mostrar
    n = i + 1
    sucesion.append(12000 + i * 2000)  # Término general
    print(f'El término {n:2d} es ${sucesion[i]:,.0f}'.replace(",", "."))


El término  1 es $12.000
El término  2 es $14.000
El término  3 es $16.000
El término  4 es $18.000
El término  5 es $20.000
El término  6 es $22.000
El término  7 es $24.000
El término  8 es $26.000
El término  9 es $28.000
El término 10 es $30.000
El término 11 es $32.000
El término 12 es $34.000


# Problema 7

Por lo general, los proyectos son organizados en varias etapas que deben completarse secuencialmente.

La fase inicial de un proyecto de software tiene una duración de $3$ meses y cada fase subsecuente reduce su duración en un $10 \%$ debido a la experiencia ganada. Determine:

1. ¿Cuánto dura la quinta fase?
2. Calcula la duración total del proyecto después de $10$ fases.

In [5]:
# Creación de sucesión geométrica con listas
sucesion = []

for i in range(10):  # 10 fases
    n = i + 1
    sucesion.append(3 * 0.9 ** (n - 1))  # Término general geométrico
    if n == 5:
        print(f'La fase {n} dura {sucesion[i]:.4f} meses')

# Cálculo de la duración total del proyecto
total = sum(sucesion)
print(f'\nLa duración total del proyecto después de 10 fases es de {total:.4f} meses')




La fase 5 dura 1.9683 meses

La duración total del proyecto después de 10 fases es de 19.5396 meses


# Problema 8

En una sucesión aritmética, la diferencia entre un término y el anterior es $4$ y el décimo término es $-20$. Determine:

1. El primer término de la sucesión.
2. La expresión del término de lugar $n$.
3. Utilice un código en *Python* para calcular el término que ocupa el lugar $100$.
4. Determinar la posición del número $1.680$ en la secuencia. Calcule utilizando un ciclo *while*.

In [ ]:
# A10= A1 + 4 (10-1)
#-20 = A1 + 4 (9)
#-20 = A1 + 36
#-56 = A1
# A1= -56

sucesion = []
for i in range(100):
    n = i + 1
    sucesion.append(-56 + 4 * (n - 1))
    print(f'El término {n} es {sucesion[i]}')

#Respuesta 1 "El termino 1 es -56"

#Respuesta 3 "El término 100 es 340""
#Respuesta 3 "El 1680 es un término de la sucesión y ocupa el lugar 435"


#Determinar si un determinado valor es término de la sucesión.
final=1680 
termino=0
n=0
dif=final-termino
while dif>0:
    n +=1
    termino=-56 + 4 * (n - 1)
    dif=final-termino

if dif==0:
    print(f'El {final} es un término de la sucesión y ocupa el lugar {n}')

else:
    print(f'El {final} no pertenece a la sucesión')

El término 1 es -56
El término 2 es -52
El término 3 es -48
El término 4 es -44
El término 5 es -40
El término 6 es -36
El término 7 es -32
El término 8 es -28
El término 9 es -24
El término 10 es -20
El término 11 es -16
El término 12 es -12
El término 13 es -8
El término 14 es -4
El término 15 es 0
El término 16 es 4
El término 17 es 8
El término 18 es 12
El término 19 es 16
El término 20 es 20
El término 21 es 24
El término 22 es 28
El término 23 es 32
El término 24 es 36
El término 25 es 40
El término 26 es 44
El término 27 es 48
El término 28 es 52
El término 29 es 56
El término 30 es 60
El término 31 es 64
El término 32 es 68
El término 33 es 72
El término 34 es 76
El término 35 es 80
El término 36 es 84
El término 37 es 88
El término 38 es 92
El término 39 es 96
El término 40 es 100
El término 41 es 104
El término 42 es 108
El término 43 es 112
El término 44 es 116
El término 45 es 120
El término 46 es 124
El término 47 es 128
El término 48 es 132
El término 49 es 136
El término

# Problema 9

El tercer término de una sucesión geométrica es $5$ y el sexto término es $40$. Determine:

1. La razón geométrica de la sucesión.
2. El primer término de la sucesión.
3. La expresión del término de lugar $n$.
4. Utilice un código en *Python* para calcular el término que ocupa el lugar $8$.
4. ¿Qué lugar ocupa en esta sucesión el término $20.480$? Calcule utilizando un ciclo *while*.

In [ ]:
# Datos iniciales
A3 = 5
A6 = 40

# 1. Cálculo de la razón geométrica (R)
R = (A6 / A3) ** (1 / (6 - 3))

# 2. Cálculo del primer término (A1)
A1 = A3 / (R ** (3 - 1))

# 3. Mostrar la expresión general del término n
print(f"Expresión general: An = {A1} * ({R})**(n - 1)")

# 4. Calcular el término de lugar 8
n = 8
A8 = A1 * (R ** (n - 1))
print(f"\nEl término 8 es: {A8}")

# 5. Determinar qué lugar ocupa el término 20.480
termino = 20480
n = 1
valor = A1
while valor < termino:
    valor = A1 * (R ** (n - 1))
    if valor == termino:
        print(f"\nEl término {termino} ocupa el lugar n = {n}")
        break
    n += 1


NameError: name 'R' is not defined

# Problema 10

Una empresa de tecnología está optimizando sus servidores. Para ello mide las transacciones por segundo (TPS) de sus servidores en cada mes. Se ha obtenido que en el segundo mes, el rendimiento del servidor fue $1,2$ miles de TPS y en el quinto mes, el rendimiento fue $2,7$ miles de TPS. Si el número de TPS que hay mes a mes sigue una sucesión geométrica, determina utilizando un código en *Python*:

1. ¿Cuál es el rendimiento el noveno mes?
2. ¿En qué mes el rendimiento es de $23,4$ miles de TPS?

Utilice dos decimales para el cálculo.

# Problema 11

Una aplicación de redes sociales se estrenó el $1$ de enero con una cierta cantidad de usuarios. A partir de ahí, el número de usuarios nuevos de cada día fue siempre el mismo. El $3$ de enero tenía $500$ usuarios en total y el $10$ de enero tenía $1.501$ usuarios totales.Determine:

1. ¿Con cuántos usuarios comenzó a funcionar la aplicación?
2. ¿Cuántos usuarios tendrá la aplicación el $31$ de enero?
3. Si la aplicación tiene $5.505$ usuarios, ¿cuántos meses lleva funcionando?



# Problema 12

Una empresa de publicidad ofrece para un puesto de trabajo un salario inicial y un aumento mensual en una razón constante. Javier se presenta a la entrevista, y el encargado del departamento de recursos humanos le explica que en el segundo mes de trabajo su sueldo será de $\$402.000$, y que en el tercer mes de trabajo será de $\$404.010$. Determine:

1. El salario aproximado que Javier recibiría en el décimo mes de trabajo en la empresa.
2. ¿Cuál sería el ingreso total recibido por los sueldos de Javier durante el tercer año de trabajo?

# Problema 13

Una empresa de ciberseguridad capacita trabajadores cada trimestre. En el primer trimestre capacitaron a $30$ trabajadores, en el segundo trimestre a $60$ trabajadores, y en el tercer trimestre a $120$ trabajadores. Si en los siguientes trimestres la empresa mantiene el mismo patrón de crecimiento, determine utilizando *Python*:

1. ¿A cuántos trabajadores capacitarán en el octavo trimestre?
2. Si la empresa lleva en el mercado $3$ años, ¿a cuántos trabajadores han capacitado en total?

# Problema 14

Al llenar un tambor con agua, se registra que durante el minuto $4$ el tambor recibe $128$ litros, y durante el minuto $9$ recibe $88$ litros. Sabiendo que en cada minuto recibe una cantidad $d$ de litros de agua menos que en el minuto anterior, determine:

1. ¿En qué instante el tambor deja de recibir agua?
2. ¿Cuántos litros recibe en total el tambor?

Realice los cálculos anteriores utilizando *Python*.